loadgen/generic_executor_test.go

```diff
@@ -128,3 +128,36 @@ func TestRunFailDuration(t *testing.T) {
 	require.ErrorContains(t, err, "run finished with error")
 	tracker.assertSeen(t, 2)
 }
+
+func TestRunDurationWithTimeout(t *testing.T) {
+	tracker := newIterationTracker()
+	err := execute(&GenericExecutor{
+		Execute: func(ctx context.Context, run *Run) error {
+			tracker.track(run.Iteration)
+			time.Sleep(time.Millisecond * 20)
+			return nil
+		},
+		DefaultConfiguration: RunConfiguration{
+			Duration: 100 * time.Millisecond,
+			Timeout:  10 * time.Millisecond},
+	})
+	require.Error(t, err)
+	require.ErrorContains(t, err, "timed out")
+}
+
+func TestRunIterationsWithTimeout(t *testing.T) {
+	tracker := newIterationTracker()
```

Comment:
Doesn't look like this tracker is asserting anything here of in the test above. You can remove it.